tests/filecheck/transforms/test-remove-memref-copy.mlir

```diff
@@ -0,0 +1,244 @@
+      %12 = arith.extsi %arg3 : i8 to i32
+      %13 = arith.subi %12, %arg5 : i32
+      %14 = arith.extsi %arg4 : i8 to i32
+      %15 = arith.subi %14, %arg6 : i32
+      %16 = arith.muli %13, %15 : i32
+      %17 = arith.addi %arg7, %16 : i32
+      linalg.yield %17 : i32
+    }
+    "memref.copy"(%11, %arg2) : (memref<?x128xi32, #tsl.tsl<[?, 8] -> (256, 4), [?, 8] -> (?, 32)>, "L1">, memref<?x128xi32, "L3">) -> ()
+    func.return %arg2 : memref<?x128xi32, "L3">
+  }
+}
+
+// CHECK: builtin.module {
+// CHECK-NEXT:   func.func @mnist(%arg0 : memref<?x128xi8, "L3">, %arg1 : memref<128x128xi8, "L3">, %arg2 : memref<?x128xi32, "L3">) -> memref<?x128xi32, "L3"> {
+// CHECK-NEXT:     %0 = arith.constant 0 : i32
+// CHECK-NEXT:     %1 = arith.constant 0 : i32
+// CHECK-NEXT:     %2 = "memref.memory_space_cast"(%arg0) : (memref<?x128xi8, "L3">) -> memref<?x128xi8, "L1">
+// CHECK-NEXT:     %3 = "memref.memory_space_cast"(%arg1) : (memref<128x128xi8, "L3">) -> memref<128x128xi8, "L1">
+// CHECK-NEXT:     %4 = "memref.memory_space_cast"(%arg2) : (memref<?x128xi32, "L3">) -> memref<?x128xi32, "L1">
+// CHECK-NEXT:     %5 = arith.constant 0 : index
+// CHECK-NEXT:     %6 = "memref.dim"(%arg0, %5) : (memref<?x128xi8, "L3">, index) -> index
+// CHECK-NEXT:     %7 = memref.alloc(%6) {"alignment" = 64 : i64} : memref<?x128xi8, #tsl.tsl<[?, 8] -> (?, 8), [?, 8] -> (256, 1)>, "L1">
+// CHECK-NEXT:     %8 = memref.alloc() {"alignment" = 64 : i64} : memref<128x128xi8, #tsl.tsl<[?, 8] -> (256, 1), [?, 8] -> (?, 8), offset: 64>, "L1">
+// CHECK-NEXT:     %9 = arith.constant 0 : index
+// CHECK-NEXT:     %10 = "memref.dim"(%arg2, %9) : (memref<?x128xi32, "L3">, index) -> index
+// CHECK-NEXT:     %11 = memref.alloc(%10) {"alignment" = 64 : i64} : memref<?x128xi32, #tsl.tsl<[?, 8] -> (256, 4), [?, 8] -> (?, 32)>, "L1">
+// CHECK-NEXT:     linalg.generic {indexing_maps = [affine_map<(d0, d1, d2) -> (d0, d2)>, affine_map<(d0, d1, d2) -> (d2, d1)>, affine_map<(d0, d1, d2) -> ()>, affine_map<(d0, d1, d2) -> ()>, affine_map<(d0, d1, d2) -> (d0, d1)>], iterator_types = ["parallel", "parallel", "reduction"]} ins(%7, %8, %0, %1 : memref<?x128xi8, #tsl.tsl<[?, 8] -> (?, 8), [?, 8] -> (256, 1)>, "L1">, memref<128x128xi8, #tsl.tsl<[?, 8] -> (256, 1), [?, 8] -> (?, 8), offset: 64>, "L1">, i32, i32) outs(%11 : memref<?x128xi32, #tsl.tsl<[?, 8] -> (256, 4), [?, 8] -> (?, 32)>, "L1">) {
+// CHECK-NEXT:     ^0(%arg3 : i8, %arg4 : i8, %arg5 : i32, %arg6 : i32, %arg7 : i32):
+// CHECK-NEXT:       %12 = arith.extsi %arg3 : i8 to i32
+// CHECK-NEXT:       %13 = arith.subi %12, %arg5 : i32
+// CHECK-NEXT:       %14 = arith.extsi %arg4 : i8 to i32
+// CHECK-NEXT:       %15 = arith.subi %14, %arg6 : i32
+// CHECK-NEXT:       %16 = arith.muli %13, %15 : i32
+// CHECK-NEXT:       %17 = arith.addi %arg7, %16 : i32
+// CHECK-NEXT:       linalg.yield %17 : i32
+// CHECK-NEXT:     }
+// CHECK-NEXT:     func.return %arg2 : memref<?x128xi32, "L3">
+// CHECK-NEXT:   }
+// CHECK-NEXT: }
+
```

Comment:
unless these are testing something specific, these seem kind of overbodig

Reply:
You are probably right :) will remove the overbodige code :)